modules/v2x/v2x_proxy/os_interface/os_interface.cc

```diff
@@ -71,8 +71,11 @@ bool OsInterFace::InitWriters() {
   v2x_traffic_light_hmi_writer_ =
       node_->CreateWriter<::apollo::perception::TrafficLightDetection>(
           FLAGS_v2x_traffic_light_for_hmi_topic);
-
-  return !!v2x_obu_traffic_light_writer_ && !!v2x_traffic_light_writer_;
+  v2x_obstacles_internal_writer_ =
+      node_->CreateWriter<apollo::v2x::V2XObstacles>(
+          FLAGS_v2x_internal_obstacle_topic);
+  return !!v2x_obu_traffic_light_writer_ && !!v2x_traffic_light_hmi_writer_ &&
+         !!v2x_obstacles_internal_writer_ && !!v2x_traffic_light_writer_;
```

Comment:
What's meaning '!!' ?

Reply:
Convert pointer to bollean instead of use lots of ```nullptr !=```

Reply:
Please removed '!!' or replaced it by 'nullptr != XXX' .

Reply:
done